autoscheduler/frontend/src/tests/ReduxAvailability.test.tsx

```diff
@@ -0,0 +1,164 @@
+import DayOfWeek from '../types/DayOfWeek';
+
+/**
+ * Converts a pair of hours and minutes into a number of minutes past midnight
+ */
+const makeTime = (h: number, m: number): number => h * 60 + m;
+/**
+ * Makes availabilities "Busy on Wednesday", since that part isn't being tested
+ */
+const dummyArgs = {
+  available: AvailabilityType.BUSY,
+  dayOfWeek: DayOfWeek.WED,
+};
+describe('Availabilities', () => {
+  describe('of the same type are merged', () => {
+    test('with an overlap on one end', () => {
+      // arrange
+      const store = createStore(autoSchedulerReducer);
+      const availability1 = {
+        ...dummyArgs,
+        time1: makeTime(8, 0),
+        time2: makeTime(12, 0),
+      };
+      const availability2 = {
+        ...dummyArgs,
+        time1: makeTime(11, 30),
+        time2: makeTime(13, 42),
+      };
+
+      // act
+      store.dispatch(addAvailability(availability1));
+      store.dispatch(addAvailability(availability2));
+
+      // assert
+      expect(store.getState().availability).toEqual([{
+        available: AvailabilityType.BUSY,
+        dayOfWeek: DayOfWeek.WED,
+        startTimeHours: 8,
+        startTimeMinutes: 0,
+        endTimeHours: 13,
+        endTimeMinutes: 42,
+      }]);
+    });
+    test('with overlaps on both ends', () => {
+      // arrange
+      const store = createStore(autoSchedulerReducer);
+      const availability1 = {
+        ...dummyArgs,
+        time1: makeTime(11, 30),
+        time2: makeTime(13, 42),
+      };
+      const availability2 = {
+        ...dummyArgs,
+        time1: makeTime(7, 0),
+        time2: makeTime(8, 30),
+      };
+      const availability3 = {
+        available: AvailabilityType.BUSY,
+        dayOfWeek: DayOfWeek.WED,
+        time1: makeTime(8, 0),
+        time2: makeTime(12, 0),
+      };
+
+      // act
+      store.dispatch(addAvailability(availability1));
+      store.dispatch(addAvailability(availability2));
+      store.dispatch(addAvailability(availability3));
+
+      // assert
+      expect(store.getState().availability).toEqual([{
+        available: AvailabilityType.BUSY,
+        dayOfWeek: DayOfWeek.WED,
+        startTimeHours: 7,
+        startTimeMinutes: 0,
+        endTimeHours: 13,
+        endTimeMinutes: 42,
+      }]);
+    });
+  });
+
+  test('that don\'t overlap are not merged', () => {
+    // arrange
+    const store = createStore(autoSchedulerReducer);
+    const availability1 = {
+      ...dummyArgs,
+      time1: makeTime(8, 0),
+      time2: makeTime(12, 0),
+    };
+    const availability2 = {
+      ...dummyArgs,
+      time1: makeTime(12, 30),
+      time2: makeTime(13, 42),
+    };
+
+    // act
+    store.dispatch(addAvailability(availability1));
+    store.dispatch(addAvailability(availability2));
+
+    // assert
+    expect(store.getState().availability).toEqual(
+      [argsToAvailability(availability1), argsToAvailability(availability2)],
+    );
+  });
+
+  test('are deleted', () => {
```

Comment:
While I can't think of any other scenarios for testing delete, I think it'd be better if this was also in its own describe and the actual test described the exact scenario you're testing, as reading it as a glance I'm not really sure what situation you're testing. How you set up the tests for merge availability is perfect in this regard.

This also applies to update, where I think it would especially benefit from explaining the scenario you're testing in the test name. 

Basically I'm trying to find the middle ground of how Jest tests are setup versus my normal naming scheme (coming from Java/C#) of `SystemUnderTest_ScenarioWe'reTesting_ExpectedResult`

Reply:
How about the top-level describe blocks names the system under test (pretty much the same as the file name), the middle describe identifies the expected result, and the test is named for the exact scenario tested?

Reply:
Yeah I think that's perfect